### ISSUE TITLE
Preload scripts don't work when in a path with non-trivial Unicode characters

### ISSUE THREAD

**Issue by cgauld** [electron/electron#10341]
* Electron version: 1.7.5
* Operating system: Both MacOS 10.12 and Windows 10

### Expected behavior

We launch a BrowserWindow with `sandbox: true` and a `preload` setting, like this:
```
mainWindow = new BrowserWindow({
    title: "...",
    webPreferences: {
        sandbox: true,
        plugins: true,
        preload: path.join(__dirname, 'preload.js'),
        'experimentalFeatures': true,
        'experimentalCanvasFeatures': true
    }
});
mainWindow.loadURL("file://" + __dirname + "/main.html");
```
This works correctly most of the time - happy days.

### Actual behavior

When the preload file (i.e. preload.js above) has a non-BMP Unicode character in its absolute path, it isn't loaded and run before the page load. Interestingly the loadURL path works fine - the window still loads main.html, just not having run preload.js first. We've observed this on both Windows and MacOS.

Trying to get to the bottom of this it looks like the preload setting is communicated to the render processes as a command line parameter (whereas loadURL goes over IPC, I think), but that's as far as I've got I'm afraid.

We also tried using a file:// path and the `preloadURL` parameter instead but with the same failure.

### How to reproduce

The easiest way to do this is to rename your project's folder to include a non-BMP Unicode character. Here's one for your copy-paste convenience -> `å`.

Thanks all - any help greatly appreciated!

**Comment by cgauld** [comment]
Thanks for labeling @sofianguy! We've also experienced this issue on MacOS :-)

**Comment by dittos** [comment]
This is fixed in 2.0.0 (#12643) but not backported to previous versions.

**Comment by codebytere** [comment]
We are no longer implementing bugfixes for versions of Electron <= `1.7.x`, so i'm going to close this issue but if it is still persisting in more recent versions of Electron we can certainly reopen it!
